monai/metrics/panoptic_quality.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+import torch
+
+from monai.metrics.utils import do_metric_reduction, remap_instance_id
+from monai.utils import MetricReduction, optional_import
+
+from .metric import CumulativeIterationMetric
+
+linear_sum_assignment, _ = optional_import("scipy.optimize", name="linear_sum_assignment")
+
+__all__ = ["PanopticQualityMetric", "compute_panoptic_quality"]
+
+
+class PanopticQualityMetric(CumulativeIterationMetric):
+    """
+    Compute panoptic quality between two tensors.
+    Input `y_pred` is compared with ground truth `y`.
+    `y_preds` and `y` should only have non-negative integer values, where each distinct value correspond to a
+    distinct instance.
+    `y_preds` and `y` can be a list of channel-first Tensor (CHW[D]) or a batch-first Tensor (BCHW[D]) and the
+    number of channel should be 1.
+
+    This class refers to the implementation in:
+    https://github.com/TissueImageAnalytics/CoNIC/blob/main/misc/utils.py
+    This class also refers to the following paper:
+    https://openaccess.thecvf.com/content_CVPR_2019/papers/Kirillov_Panoptic_Segmentation_CVPR_2019_paper.pdf
+
+    Example of the typical execution steps of this metric class follows :py:class:`monai.metrics.metric.Cumulative`.
+
+    Args:
+        reduction: define mode of reduction to the metrics,
+            available reduction modes: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``},
+            default to ``"mean"``. if "none", will not do reduction.
+        remap: whether to remap `y_pred` and `y` to ensure contiguous ordering of instance id.
+        match_iou: IOU threshould to determine the paring between `y_pred` and `y`. Usually,
```

Comment:
"match_iou: IOU threshould to determine the **pairing** between `y_pred` and `y`. Usually,"
There are a couple of other instances where **paring** is used instead of **pairing**